pynicotine/shares.py

```diff
@@ -112,50 +80,192 @@ class FileTypes:
+
+                try:
+                    file_header_length = current_offset = len(self.FILE_HEADER)
+
+                    if contents_view[:file_header_length] != self.FILE_HEADER:
+                        raise DatabaseError("Not a database file")
+
+                    while current_offset < file_size:
+                        key_offset = (current_offset + self.LENGTH_DATA_SIZE)
+                        key_length, value_length = self.UNPACK_LENGTHS(contents_view[current_offset:key_offset])
+                        value_offset = (key_offset + key_length)
+                        key = str(contents_view[key_offset:value_offset], encoding="utf-8")
+
+                        value_offsets[key] = value_offset
+                        current_offset = (value_offset + value_length)
+                finally:
+                    contents_view.release()
+
+        return value_offsets
+
+    def __contains__(self, key):
+        return key in self._value_offsets
+
+    def __iter__(self):
+        for key in self._value_offsets:
+            yield key
+
+    def __len__(self):
+        return len(self._value_offsets)
+
+    def __getitem__(self, key):
+
+        value_offset = self._value_offsets[key]
+
+        self._file_handle.seek(value_offset, SEEK_SET)
+        return RestrictedUnpickler(self._file_handle).load()
+
+    def __setitem__(self, key, value):
+
+        encoded_key = str(key).encode("utf-8")
+        pickled_value = dumps(value, protocol=HIGHEST_PROTOCOL)
+
+        key_length = len(encoded_key)
+        length_data = self.PACK_LENGTHS(key_length, len(pickled_value))
+        item_data = (length_data + encoded_key + pickled_value)
+
+        self._file_handle.write(item_data)
+
+        self._value_offsets[key] = self._file_offset + self.LENGTH_DATA_SIZE + key_length
+        self._file_offset += len(item_data)
+
+    def get(self, key, default=None):
+
+        if key in self._value_offsets:
+            return self[key]
+
+        return default
+
+    def update(self, obj):
+        for key, value in obj.items():
+            self[key] = value
+
+    def close(self):
+
+        if self._overwrite:
+            os.fsync(self._file_handle)
+
+        self._file_handle.close()
+
+
 class Scanner:
     """Separate process responsible for building shares.
 
     It handles scanning of folders and files, as well as building
     databases and writing them to disk.
     """
 
-    def __init__(self, config_obj, queue, shared_folders, share_db_paths, init=False, rescan=True, rebuild=False):
+    def __init__(self, config_obj, queue, shared_folders, share_db_paths, init=False, rescan=True,
+                 rebuild=False, is_buddy_visible=False, is_trusted_visible=False):
 
         self.config = config_obj
         self.queue = queue
-        self.shared_folders, self.shared_buddy_folders = shared_folders
+        self.shared_folders, self.shared_buddy_folders, self.shared_trusted_folders = shared_folders
```

Comment:
Consider
`self.shared_folders["public"]`, `self.shared_folders["buddy"]`, `self.shared_folders["trusted"]`
could help to make simplifications elsewhere in this module by referencing `self.shared_folders[share_group]`.